faster_sam/cloudformation.py

```diff
@@ -9,28 +9,72 @@
 
 
 class CFTemplateNotFound(FileNotFoundError):
+    """Raised when the CloudFormation template file cannot be found."""
+
     pass
 
 
 class CFBadTag(TypeError):
+    """Raised when an invalid CloudFormation tag is encountered."""
+
     pass
 
 
 class CFBadNode(ValueError):
+    """Raised when an invalid CloudFormation node is encountered."""
+
     pass
 
 
 class CFLoader(yaml.SafeLoader):
+    """Custom YAML loader for CloudFormation templates."""
+
     pass
 
 
 class NodeType(Enum):
+    """
+    Enum representing different types of CloudFormation nodes.
+
+    Attributes
+    ----------
+    API_GATEWAY : str
+        Represents the "AWS::Serverless::Api" node type.
+    LAMBDA : str
+        Represents the "AWS::Serverless::Function" node type.
+    API_EVENT : str
+        Represents the "Api" node type.
+    """
+
     API_GATEWAY = "AWS::Serverless::Api"
     LAMBDA = "AWS::Serverless::Function"
     API_EVENT = "Api"
 
 
 def multi_constructor(loader: CFLoader, tag_suffix: str, node: yaml.nodes.Node) -> Dict[str, Any]:
+    """
+    Multi-constructor function for handling different CloudFormation template loading.
```

Comment:
Sugestão para a descrição desta função:

_Custom YAML node constructor._

_Handles AWS CloudFormation extensions for its short version of intrinsic functions._